lib/Sema/TypeCheckAttr.cpp

```diff
@@ -119,6 +119,7 @@ class AttributeChecker : public AttributeVisitor<AttributeChecker> {
   IGNORED_ATTR(ImplicitSelfCapture)
   IGNORED_ATTR(InheritActorContext)
   IGNORED_ATTR(Isolated)
+  IGNORED_ATTR(UnavailableFromAsync)
```

Comment:
... here's where I think we should do the "you can't put this attribute on an `async` declaration" check.